From 28099aa0a5cce439b18e5ac2365a4d6e5ab045da Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Tue, 14 Jul 2020 16:30:10 +0800
Subject: [PATCH 1/3] soc: fsl: rcpm: move out the while condition judgment to
 avoid dead loop

The function rcpm_pm_prepare has continue operation when do while loop,
This will cause the while condition not to be set and always keep the
true, so move out the while condition judgment to avoid dead loop.

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/soc/fsl/rcpm.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/soc/fsl/rcpm.c b/drivers/soc/fsl/rcpm.c
index 73d5bbb75cbf..81e527f94bda 100644
--- a/drivers/soc/fsl/rcpm.c
+++ b/drivers/soc/fsl/rcpm.c
@@ -98,8 +98,7 @@ static int rcpm_pm_prepare(struct device *dev)
 				}
 			}
 		}
-		ws = wakeup_source_get_next(ws);
-	} while (ws != NULL);
+	} while ((ws = wakeup_source_get_next(ws)));
 
 	wakeup_source_get_stop(idx);
 
-- 
2.17.1

